rlib-format-packed-bundled-libs test failures on Xcode 14

The tests `src/test/run-make/rlib-format-packed-bundled-libs` and `src/test/run-make/rlib-format-packed-bundled-libs-2` are failing with Xcode 14. The output is:

```
running 2 tests
FF
failures:

---- [run-make] src/test/run-make/rlib-format-packed-bundled-libs-2 stdout ----

error: make failed
status: exit status: 2
command: "make"
--- stdout -------------------------------
# Build strange-named dep.
DYLD_LIBRARY_PATH="/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2:/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib:" '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/bin/rustc' --out-dir /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2 -L /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2  native_dep.rs --crate-type=staticlib -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/native_dep.ext
DYLD_LIBRARY_PATH="/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2:/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib:" '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/bin/rustc' --out-dir /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2 -L /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2  rust_dep.rs --crate-type=rlib -Zpacked_bundled_libs
"/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin"/llvm-nm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/librust_dep.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "U.*native_f1"
[[[ begin stdout ]]]

lib.rmeta:

rust_dep.rust_dep.f8908835-cgu.0.rcgu.o:
                 U __ZN4core9panicking5panic17h2cab1196791b6748E
---------------- T __ZN8rust_dep8rust_dep17h46da1ffdf057d344E
                 U _native_f1

[[[ end stdout ]]]
ar t /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/librust_dep.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" "native_dep.ext"
[[[ begin stdout ]]]
__.SYMDEF
lib.rmeta
rust_dep.rust_dep.f8908835-cgu.0.rcgu.o
native_dep.ext

[[[ end stdout ]]]
# Make sure compiler doesn't use files, that it shouldn't know about.
rm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/native_dep.ext
DYLD_LIBRARY_PATH="/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2:/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib:" '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/bin/rustc' --out-dir /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2 -L /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2  main.rs --extern rust_dep=/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/librust_dep.rlib -Zpacked_bundled_libs
------------------------------------------
--- stderr -------------------------------
warning: ignoring --out-dir flag due to -o flag

warning: 1 warning emitted

/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/librust_dep.rlib:lib.rmeta: no symbols
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-arch" "x86_64" "-m64" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/rustciJPqdZ/symbols.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/main.main.cbcd4161-cgu.0.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/main.main.cbcd4161-cgu.1.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/main.main.cbcd4161-cgu.2.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/main.main.cbcd4161-cgu.3.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/main.main.cbcd4161-cgu.4.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/main.4g86d6gt786od2ny.rcgu.o" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/librust_dep.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/rustciJPqdZ/native_dep.ext" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libstd-258e71d5a19fa69d.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-5b1ca701534fabd3.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libobject-aa5f0fc465004fb6.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libmemchr-3d0c3dc0ed2a3ca7.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-a2f5bcebde3d6ba1.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libgimli-1c169b69cac8e3f9.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-4a5c1257c19a64c5.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libstd_detect-53c2b17aef5da639.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-f0d1d36275bd5157.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-26d526555da8e92a.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libminiz_oxide-6ac28b1a995b7da1.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libadler-e2ee1886e035c392.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-83a804165d411a51.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libunwind-9cde8e1ff18e7da9.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-1f3b7d2631393666.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/liblibc-18eaf45b9a0638b0.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/liballoc-44a61fa47279665f.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-9f743963c507336d.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcore-3e6d7d0b5b99b826.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-d7fc134a6ac5d183.rlib" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/main" "-Wl,-dead_strip" "-nodefaultlibs"
  = note: ld: in /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/librust_dep.rlib(native_dep.ext), archive member 'native_dep.ext' with length 25521328 is not mach-o or llvm bitcode file '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs-2/rlib-format-packed-bundled-libs-2/librust_dep.rlib'
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

make: *** [all] Error 1
------------------------------------------


---- [run-make] src/test/run-make/rlib-format-packed-bundled-libs stdout ----

error: make failed
status: exit status: 2
command: "make"
--- stdout -------------------------------
cc -ffunction-sections -fdata-sections -fPIC -m64 -arch x86_64 -stdlib=libc++ -v -c -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_1.o native_dep_1.c
ar crus /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_1.a /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_1.o
cc -ffunction-sections -fdata-sections -fPIC -m64 -arch x86_64 -stdlib=libc++ -v -c -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_2.o native_dep_2.c
ar crus /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_2.a /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_2.o
cc -ffunction-sections -fdata-sections -fPIC -m64 -arch x86_64 -stdlib=libc++ -v -c -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_3.o native_dep_3.c
ar crus /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_3.a /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_3.o
DYLD_LIBRARY_PATH="/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs:/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib:" '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/bin/rustc' --out-dir /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs -L /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs  rust_dep_up.rs --crate-type=rlib -Zpacked_bundled_libs
"/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin"/llvm-nm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "U.*native_f2"
[[[ begin stdout ]]]

lib.rmeta:

rust_dep_up.rust_dep_up.9cf8be89-cgu.0.rcgu.o:
---------------- T __ZN11rust_dep_up11rust_dep_up17h591fcb77681d961dE
                 U __ZN4core9panicking5panic17h2cab1196791b6748E
                 U _native_f2
                 U _native_f3

[[[ end stdout ]]]
"/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin"/llvm-nm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "U.*native_f3"
[[[ begin stdout ]]]

lib.rmeta:

rust_dep_up.rust_dep_up.9cf8be89-cgu.0.rcgu.o:
---------------- T __ZN11rust_dep_up11rust_dep_up17h591fcb77681d961dE
                 U __ZN4core9panicking5panic17h2cab1196791b6748E
                 U _native_f2
                 U _native_f3

[[[ end stdout ]]]
"/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin"/llvm-nm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "T.*rust_dep_up"
[[[ begin stdout ]]]

lib.rmeta:

rust_dep_up.rust_dep_up.9cf8be89-cgu.0.rcgu.o:
---------------- T __ZN11rust_dep_up11rust_dep_up17h591fcb77681d961dE
                 U __ZN4core9panicking5panic17h2cab1196791b6748E
                 U _native_f2
                 U _native_f3

[[[ end stdout ]]]
ar t /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" "native_dep_2"
[[[ begin stdout ]]]
__.SYMDEF
lib.rmeta
rust_dep_up.rust_dep_up.9cf8be89-cgu.0.rcgu.o
libnative_dep_2.a
libnative_dep_3.a

[[[ end stdout ]]]
ar t /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" "native_dep_3"
[[[ begin stdout ]]]
__.SYMDEF
lib.rmeta
rust_dep_up.rust_dep_up.9cf8be89-cgu.0.rcgu.o
libnative_dep_2.a
libnative_dep_3.a

[[[ end stdout ]]]
DYLD_LIBRARY_PATH="/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs:/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib:" '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/bin/rustc' --out-dir /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs -L /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs  rust_dep_local.rs --extern rlib=/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib -Zpacked_bundled_libs --crate-type=rlib
"/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin"/llvm-nm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "U.*native_f1"
[[[ begin stdout ]]]

lib.rmeta:

rust_dep_local.rust_dep_local.17a839ca-cgu.0.rcgu.o:
                 U __ZN11rust_dep_up11rust_dep_up17h591fcb77681d961dE
---------------- T __ZN14rust_dep_local14rust_dep_local17h9562439c45fb1788E
                 U __ZN4core9panicking5panic17h2cab1196791b6748E
                 U _native_f1

[[[ end stdout ]]]
"/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin"/llvm-nm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "T.*rust_dep_local"
[[[ begin stdout ]]]

lib.rmeta:

rust_dep_local.rust_dep_local.17a839ca-cgu.0.rcgu.o:
                 U __ZN11rust_dep_up11rust_dep_up17h591fcb77681d961dE
---------------- T __ZN14rust_dep_local14rust_dep_local17h9562439c45fb1788E
                 U __ZN4core9panicking5panic17h2cab1196791b6748E
                 U _native_f1

[[[ end stdout ]]]
ar t /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" "native_dep_1"
[[[ begin stdout ]]]
__.SYMDEF
lib.rmeta
rust_dep_local.rust_dep_local.17a839ca-cgu.0.rcgu.o
libnative_dep_1.a

[[[ end stdout ]]]
# Make sure compiler doesn't use files, that it shouldn't know about.
rm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/*native_dep_*
DYLD_LIBRARY_PATH="/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs:/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib:" '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/bin/rustc' --out-dir /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs -L /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs  main.rs --extern lib=/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.exe -Zpacked_bundled_libs --print link-args | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "native_dep_1.*native_dep_2.*native_dep_3"
[[[ begin stdout ]]]
"cc" "-arch" "x86_64" "-m64" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/symbols.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.0.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.1.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.2.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.3.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.4.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.4g86d6gt786od2ny.rcgu.o" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/libnative_dep_1.a" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/libnative_dep_2.a" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/libnative_dep_3.a" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libstd-258e71d5a19fa69d.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-5b1ca701534fabd3.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libobject-aa5f0fc465004fb6.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libmemchr-3d0c3dc0ed2a3ca7.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-a2f5bcebde3d6ba1.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libgimli-1c169b69cac8e3f9.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-4a5c1257c19a64c5.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libstd_detect-53c2b17aef5da639.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-f0d1d36275bd5157.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-26d526555da8e92a.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libminiz_oxide-6ac28b1a995b7da1.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libadler-e2ee1886e035c392.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-83a804165d411a51.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libunwind-9cde8e1ff18e7da9.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-1f3b7d2631393666.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/liblibc-18eaf45b9a0638b0.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/liballoc-44a61fa47279665f.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-9f743963c507336d.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcore-3e6d7d0b5b99b826.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-d7fc134a6ac5d183.rlib" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.exe" "-Wl,-dead_strip" "-nodefaultlibs"

[[[ end stdout ]]]
"/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin"/llvm-nm /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.exe | "/Users/eric/Proj/rust/rust/src/etc/cat-and-grep.sh" -e "T.*native_f1"
[[[ begin stdout ]]]

[[[ end stdout ]]]
Error: cannot match: T.*native_f1
------------------------------------------
--- stderr -------------------------------
Apple clang version 14.0.0 (clang-1400.0.29.102)
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple x86_64-apple-macosx12.0.0 -Wundef-prefix=TARGET_OS_ -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name native_dep_1.c -mrelocation-model pic -pic-level 2 -mframe-pointer=all -fno-strict-return -fno-rounding-math -funwind-tables=2 -target-sdk-version=12.3 -fvisibility-inlines-hidden-static-local-var -target-cpu penryn -tune-cpu generic -debugger-tuning=lldb -target-linker-version 819.6 -v -ffunction-sections -fdata-sections -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -I/usr/local/include -internal-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include -internal-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -Wno-reserved-identifier -Wno-gnu-folding-constant -Wno-cast-function-type -Wno-bitwise-instead-of-logical -fdebug-compilation-dir=/Users/eric/Proj/rust/rust/src/test/run-make/rlib-format-packed-bundled-libs -ferror-limit 19 -stack-protector 1 -fstack-check -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fmax-type-align=16 -fcommon -clang-vendor-feature=+messageToSelfInClassMethodIdReturnType -clang-vendor-feature=+disableInferNewAvailabilityFromInit -clang-vendor-feature=+disableNonDependentMemberExprInCurrentInstantiation -fno-odr-hash-protocols -clang-vendor-feature=+enableAggressiveVLAFolding -clang-vendor-feature=+revert09abecef7bbf -clang-vendor-feature=+thisNoAlignAttr -clang-vendor-feature=+thisNoNullAttr -mllvm -disable-aligned-alloc-awareness=1 -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_1.o -x c native_dep_1.c
clang -cc1 version 14.0.0 (clang-1400.0.29.102) default target x86_64-apple-darwin21.6.0
ignoring nonexistent directory "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
Apple clang version 14.0.0 (clang-1400.0.29.102)
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple x86_64-apple-macosx12.0.0 -Wundef-prefix=TARGET_OS_ -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name native_dep_2.c -mrelocation-model pic -pic-level 2 -mframe-pointer=all -fno-strict-return -fno-rounding-math -funwind-tables=2 -target-sdk-version=12.3 -fvisibility-inlines-hidden-static-local-var -target-cpu penryn -tune-cpu generic -debugger-tuning=lldb -target-linker-version 819.6 -v -ffunction-sections -fdata-sections -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -I/usr/local/include -internal-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include -internal-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -Wno-reserved-identifier -Wno-gnu-folding-constant -Wno-cast-function-type -Wno-bitwise-instead-of-logical -fdebug-compilation-dir=/Users/eric/Proj/rust/rust/src/test/run-make/rlib-format-packed-bundled-libs -ferror-limit 19 -stack-protector 1 -fstack-check -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fmax-type-align=16 -fcommon -clang-vendor-feature=+messageToSelfInClassMethodIdReturnType -clang-vendor-feature=+disableInferNewAvailabilityFromInit -clang-vendor-feature=+disableNonDependentMemberExprInCurrentInstantiation -fno-odr-hash-protocols -clang-vendor-feature=+enableAggressiveVLAFolding -clang-vendor-feature=+revert09abecef7bbf -clang-vendor-feature=+thisNoAlignAttr -clang-vendor-feature=+thisNoNullAttr -mllvm -disable-aligned-alloc-awareness=1 -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_2.o -x c native_dep_2.c
clang -cc1 version 14.0.0 (clang-1400.0.29.102) default target x86_64-apple-darwin21.6.0
ignoring nonexistent directory "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
Apple clang version 14.0.0 (clang-1400.0.29.102)
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple x86_64-apple-macosx12.0.0 -Wundef-prefix=TARGET_OS_ -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name native_dep_3.c -mrelocation-model pic -pic-level 2 -mframe-pointer=all -fno-strict-return -fno-rounding-math -funwind-tables=2 -target-sdk-version=12.3 -fvisibility-inlines-hidden-static-local-var -target-cpu penryn -tune-cpu generic -debugger-tuning=lldb -target-linker-version 819.6 -v -ffunction-sections -fdata-sections -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -I/usr/local/include -internal-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include -internal-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -internal-externc-isystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -Wno-reserved-identifier -Wno-gnu-folding-constant -Wno-cast-function-type -Wno-bitwise-instead-of-logical -fdebug-compilation-dir=/Users/eric/Proj/rust/rust/src/test/run-make/rlib-format-packed-bundled-libs -ferror-limit 19 -stack-protector 1 -fstack-check -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fmax-type-align=16 -fcommon -clang-vendor-feature=+messageToSelfInClassMethodIdReturnType -clang-vendor-feature=+disableInferNewAvailabilityFromInit -clang-vendor-feature=+disableNonDependentMemberExprInCurrentInstantiation -fno-odr-hash-protocols -clang-vendor-feature=+enableAggressiveVLAFolding -clang-vendor-feature=+revert09abecef7bbf -clang-vendor-feature=+thisNoAlignAttr -clang-vendor-feature=+thisNoNullAttr -mllvm -disable-aligned-alloc-awareness=1 -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/libnative_dep_3.o -x c native_dep_3.c
clang -cc1 version 14.0.0 (clang-1400.0.29.102) default target x86_64-apple-darwin21.6.0
ignoring nonexistent directory "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib:lib.rmeta: no symbols
/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib:lib.rmeta: no symbols
/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib:lib.rmeta: no symbols
/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib:lib.rmeta: no symbols
/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib:lib.rmeta: no symbols
warning: ignoring --out-dir flag due to -o flag

error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-arch" "x86_64" "-m64" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/symbols.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.0.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.1.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.2.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.3.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.main.cbcd4161-cgu.4.rcgu.o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.4g86d6gt786od2ny.rcgu.o" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/libnative_dep_1.a" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_up.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/libnative_dep_2.a" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/rustchI2eOT/libnative_dep_3.a" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libstd-258e71d5a19fa69d.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-5b1ca701534fabd3.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libobject-aa5f0fc465004fb6.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libmemchr-3d0c3dc0ed2a3ca7.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-a2f5bcebde3d6ba1.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libgimli-1c169b69cac8e3f9.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-4a5c1257c19a64c5.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libstd_detect-53c2b17aef5da639.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-f0d1d36275bd5157.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-26d526555da8e92a.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libminiz_oxide-6ac28b1a995b7da1.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libadler-e2ee1886e035c392.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-83a804165d411a51.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libunwind-9cde8e1ff18e7da9.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-1f3b7d2631393666.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/liblibc-18eaf45b9a0638b0.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/liballoc-44a61fa47279665f.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-9f743963c507336d.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcore-3e6d7d0b5b99b826.rlib" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-d7fc134a6ac5d183.rlib" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.exe" "-Wl,-dead_strip" "-nodefaultlibs"
  = note: ld: in /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib(libnative_dep_1.a), archive member 'libnative_dep_1.a' with length 832 is not mach-o or llvm bitcode file '/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/librust_dep_local.rlib'
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error; 1 warning emitted

/Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/ci-llvm/bin/llvm-nm: error: /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/test/run-make/rlib-format-packed-bundled-libs/rlib-format-packed-bundled-libs/main.exe: No such file or directory
make: *** [all] Error 1
------------------------------------------



failures:
    [run-make] src/test/run-make/rlib-format-packed-bundled-libs
    [run-make] src/test/run-make/rlib-format-packed-bundled-libs-2

test result: FAILED. 0 passed; 2 failed; 0 ignored; 0 measured; 68 filtered out; finished in 4.69s
```

I'm not sure if it is helpful, but this error is curiously similar to the error that versions older than rust 1.54 produce when trying to build *any* program with Xcode 14. That was fixed via #84449.
See also https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/.E2.9C.94.20How.20can.20I.20fix.20Rust.201.2E53.2E0.20or.20earlier.20to.20run.20on.20macOS.2012.2E6.3F/near/299263887 where @bjorn3 gave a little more context on that.

cc @BelovDV who added these tests in #100101.

Marking as P-critical for now because GitHub is updating `macos-latest` to `macos-12` *now*, intending to complete by December 1. If that transition hits `rust-lang-ci`, then CI will no longer pass. See https://github.com/actions/runner-images/issues/6384 for more information. The priority can be lowered with workarounds such as manually setting `macos-11` in CI or switching to an older Xcode, or disabling these tests on macOS. If it is a simple fix, that would be ideal to try to get that in quickly.

### Meta
This is on latest master 6b3ede3f7bc502eba7bbd202b4b9312d812adcd7.

--- a/compiler/rustc_codegen_ssa/src/back/archive.rs
+++ b/compiler/rustc_codegen_ssa/src/back/archive.rs
@@ -4,8 +4,11 @@ use rustc_session::cstore::DllImport;
 use rustc_session::Session;
 use rustc_span::symbol::Symbol;
 
+use super::metadata::search_for_section;
+
 use object::read::archive::ArchiveFile;
 
+use std::error::Error;
 use std::fs::File;
 use std::io;
 use std::path::{Path, PathBuf};
@@ -56,6 +59,9 @@ pub trait ArchiveBuilderBuilder {
             if !bundled_lib_file_names.contains(&Symbol::intern(name)) {
                 continue; // We need to extract only native libraries.
             }
+            let data = search_for_section(rlib, data, ".bundled_lib").map_err(|e| {
+                ExtractBundledLibsError::ExtractSection { rlib, error: Box::<dyn Error>::from(e) }
+            })?;
             std::fs::write(&outdir.join(&name), data)
                 .map_err(|e| ExtractBundledLibsError::WriteFile { rlib, error: Box::new(e) })?;
         }

--- a/compiler/rustc_codegen_ssa/src/back/link.rs
+++ b/compiler/rustc_codegen_ssa/src/back/link.rs
@@ -8,7 +8,7 @@ use rustc_errors::{ErrorGuaranteed, Handler};
 use rustc_fs_util::fix_windows_verbatim_for_gcc;
 use rustc_hir::def_id::CrateNum;
 use rustc_metadata::find_native_static_library;
-use rustc_metadata::fs::{emit_metadata, METADATA_FILENAME};
+use rustc_metadata::fs::{emit_wrapper_file, METADATA_FILENAME};
 use rustc_middle::middle::dependency_format::Linkage;
 use rustc_middle::middle::exported_symbols::SymbolExportKind;
 use rustc_session::config::{self, CFGuard, CrateType, DebugInfo, LdImpl, Lto, Strip};
@@ -29,7 +29,7 @@ use rustc_target::spec::{RelocModel, RelroLevel, SanitizerSet, SplitDebuginfo, T
 use super::archive::{ArchiveBuilder, ArchiveBuilderBuilder};
 use super::command::Command;
 use super::linker::{self, Linker};
-use super::metadata::{create_rmeta_file, MetadataPosition};
+use super::metadata::{create_wrapper_file, MetadataPosition};
 use super::rpath::{self, RPathConfig};
 use crate::{
     errors, looks_like_rust_object_file, CodegenResults, CompiledModule, CrateInfo, NativeLib,
@@ -44,7 +44,7 @@ use std::borrow::Borrow;
 use std::cell::OnceCell;
 use std::collections::BTreeSet;
 use std::ffi::OsString;
-use std::fs::{File, OpenOptions};
+use std::fs::{read, File, OpenOptions};
 use std::io::{BufWriter, Write};
 use std::ops::Deref;
 use std::path::{Path, PathBuf};
@@ -292,8 +292,8 @@ fn link_rlib<'a>(
     let trailing_metadata = match flavor {
         RlibFlavor::Normal => {
             let (metadata, metadata_position) =
-                create_rmeta_file(sess, codegen_results.metadata.raw_data());
-            let metadata = emit_metadata(sess, &metadata, tmpdir);
+                create_wrapper_file(sess, b".rmeta".to_vec(), codegen_results.metadata.raw_data());
+            let metadata = emit_wrapper_file(sess, &metadata, tmpdir, METADATA_FILENAME);
             match metadata_position {
                 MetadataPosition::First => {
                     // Most of the time metadata in rlib files is wrapped in a "dummy" object
@@ -376,12 +376,18 @@ fn link_rlib<'a>(
             let location =
                 find_native_static_library(name.as_str(), lib.verbatim, &lib_search_paths, sess);
             if sess.opts.unstable_opts.packed_bundled_libs && flavor == RlibFlavor::Normal {
-                packed_bundled_libs.push(find_native_static_library(
-                    lib.filename.unwrap().as_str(),
+                let filename = lib.filename.unwrap();
+                let lib_path = find_native_static_library(
+                    filename.as_str(),
                     Some(true),
                     &lib_search_paths,
                     sess,
-                ));
+                );
+                let src = read(lib_path)
+                    .map_err(|e| sess.emit_fatal(errors::ReadFileError { message: e }))?;
+                let (data, _) = create_wrapper_file(sess, b".bundled_lib".to_vec(), &src);
+                let wrapper_file = emit_wrapper_file(sess, &data, tmpdir, filename.as_str());
+                packed_bundled_libs.push(wrapper_file);
                 continue;
             }
             ab.add_archive(&location, Box::new(|_| false)).unwrap_or_else(|error| {

--- a/compiler/rustc_codegen_ssa/src/back/metadata.rs
+++ b/compiler/rustc_codegen_ssa/src/back/metadata.rs
@@ -60,7 +60,7 @@ impl MetadataLoader for DefaultMetadataLoader {
                     let data = entry
                         .data(data)
                         .map_err(|e| format!("failed to parse rlib '{}': {}", path.display(), e))?;
-                    return search_for_metadata(path, data, ".rmeta");
+                    return search_for_section(path, data, ".rmeta");
                 }
             }
 
@@ -69,11 +69,11 @@ impl MetadataLoader for DefaultMetadataLoader {
     }
 
     fn get_dylib_metadata(&self, _target: &Target, path: &Path) -> Result<MetadataRef, String> {
-        load_metadata_with(path, |data| search_for_metadata(path, data, ".rustc"))
+        load_metadata_with(path, |data| search_for_section(path, data, ".rustc"))
     }
 }
 
-fn search_for_metadata<'a>(
+pub(super) fn search_for_section<'a>(
     path: &Path,
     bytes: &'a [u8],
     section: &str,
@@ -223,7 +223,11 @@ pub enum MetadataPosition {
 // * ELF - All other targets are similar to Windows in that there's a
 //   `SHF_EXCLUDE` flag we can set on sections in an object file to get
 //   automatically removed from the final output.
-pub fn create_rmeta_file(sess: &Session, metadata: &[u8]) -> (Vec<u8>, MetadataPosition) {
+pub fn create_wrapper_file(
+    sess: &Session,
+    section_name: Vec<u8>,
+    data: &[u8],
+) -> (Vec<u8>, MetadataPosition) {
     let Some(mut file) = create_object_file(sess) else {
         // This is used to handle all "other" targets. This includes targets
         // in two categories:
@@ -241,11 +245,11 @@ pub fn create_rmeta_file(sess: &Session, metadata: &[u8]) -> (Vec<u8>, MetadataP
         // WebAssembly and for targets not supported by the `object` crate
         // yet it means that work will need to be done in the `object` crate
         // to add a case above.
-        return (metadata.to_vec(), MetadataPosition::Last);
+        return (data.to_vec(), MetadataPosition::Last);
     };
     let section = file.add_section(
         file.segment_name(StandardSegment::Debug).to_vec(),
-        b".rmeta".to_vec(),
+        section_name,
         SectionKind::Debug,
     );
     match file.format() {
@@ -259,7 +263,7 @@ pub fn create_rmeta_file(sess: &Session, metadata: &[u8]) -> (Vec<u8>, MetadataP
         }
         _ => {}
     };
-    file.append_section_data(section, metadata, 1);
+    file.append_section_data(section, data, 1);
     (file.write().unwrap(), MetadataPosition::First)
 }
 

--- a/compiler/rustc_codegen_ssa/src/errors.rs
+++ b/compiler/rustc_codegen_ssa/src/errors.rs
@@ -507,6 +507,9 @@ pub enum ExtractBundledLibsError<'a> {
 
     #[diag(codegen_ssa_extract_bundled_libs_write_file)]
     WriteFile { rlib: &'a Path, error: Box<dyn std::error::Error> },
+
+    #[diag(codegen_ssa_extract_bundled_libs_write_file)]
+    ExtractSection { rlib: &'a Path, error: Box<dyn std::error::Error> },
 }
 
 #[derive(Diagnostic)]
@@ -520,4 +523,10 @@ pub struct UnsupportedArch<'a> {
 pub enum AppleSdkRootError<'a> {
     #[diag(codegen_ssa_apple_sdk_error_sdk_path)]
     SdkPath { sdk_name: &'a str, error: Error },
+}
+
+#[derive(Diagnostic)]
+#[diag(codegen_ssa_read_file)]
+pub struct ReadFileError {
+    pub message: std::io::Error,
 }

--- a/compiler/rustc_error_messages/locales/en-US/codegen_ssa.ftl
+++ b/compiler/rustc_error_messages/locales/en-US/codegen_ssa.ftl
@@ -182,3 +182,5 @@ codegen_ssa_extract_bundled_libs_write_file = failed to write file '{$rlib}': {$
 codegen_ssa_unsupported_arch = unsupported arch `{$arch}` for os `{$os}`
 
 codegen_ssa_apple_sdk_error_sdk_path = failed to get {$sdk_name} SDK path: {error}
+
+codegen_ssa_read_file = failed to read file: {message}

--- a/compiler/rustc_metadata/src/fs.rs
+++ b/compiler/rustc_metadata/src/fs.rs
@@ -22,9 +22,14 @@ pub const METADATA_FILENAME: &str = "lib.rmeta";
 /// building an `.rlib` (stomping over one another), or writing an `.rmeta` into a
 /// directory being searched for `extern crate` (observing an incomplete file).
 /// The returned path is the temporary file containing the complete metadata.
-pub fn emit_metadata(sess: &Session, metadata: &[u8], tmpdir: &MaybeTempDir) -> PathBuf {
-    let out_filename = tmpdir.as_ref().join(METADATA_FILENAME);
-    let result = fs::write(&out_filename, metadata);
+pub fn emit_wrapper_file(
+    sess: &Session,
+    data: &[u8],
+    tmpdir: &MaybeTempDir,
+    name: &str,
+) -> PathBuf {
+    let out_filename = tmpdir.as_ref().join(name);
+    let result = fs::write(&out_filename, data);
 
     if let Err(err) = result {
         sess.emit_fatal(FailedWriteError { filename: out_filename, err });

--- a/compiler/rustc_metadata/src/lib.rs
+++ b/compiler/rustc_metadata/src/lib.rs
@@ -41,6 +41,6 @@ pub mod errors;
 pub mod fs;
 pub mod locator;
 
-pub use fs::{emit_metadata, METADATA_FILENAME};
+pub use fs::{emit_wrapper_file, METADATA_FILENAME};
 pub use native_libs::find_native_static_library;
 pub use rmeta::{encode_metadata, EncodedMetadata, METADATA_HEADER};

